typescript/sdk/src/providers/transactions/submitter/builder/types.ts

```diff
@@ -15,77 +15,89 @@ export const SubmissionStrategySchema = z
 
 export type SubmissionStrategy = z.infer<typeof SubmissionStrategySchema>;
 
-export const ChainSubmissionStrategySchema = z.preprocess(
+export function preprocessChainSubmissionStrategy<
+  T extends { submitter: { type: string } },
```

Comment:
This only extends 
`{ submitter: { type: string } }` 
instead of 
`{ submitter: SubmitterMetadata }`

because SubmitterMetadata is actually a union, which will get wider as the caller extends it, leading to type errors. alternatively, we can use `{ submitter: any }`